Documentation/ExtensionArchitecture/Extbase/Reference/FileUpload.rst

```diff
@@ -0,0 +1,502 @@
+
+Each uploaded file can be validated against a configurable set of validators.
+The :php:`validation` section of the :php-short:`\TYPO3\CMS\Extbase\Annotation\FileUpload` attribute allows to
+configure commonly used validators using a configuration shorthand.
+
+The following validation rules can be configured in the :php:`validation`
+section of the :php-short:`\TYPO3\CMS\Extbase\Annotation\FileUpload` attribute:
+
+*   :php:`required`
+*   :php:`minFiles`
+*   :php:`maxFiles`
+*   :php:`fileSize`
+*   :php:`allowedMimeTypes`
+*   :php:`imageDimensions`
+
+Example:
+
+..  code-block:: php
+    :caption: Excerpt of an attribute withhin an Extbase domain model class
+
+    #[FileUpload([
+        'validation' => [
+            'required' => true,
+            'maxFiles' => 1,
+            'fileSize' => ['minimum' => '0K', 'maximum' => '2M'],
+            'allowedMimeTypes' => ['image/jpeg'],
+            'imageDimensions' => ['maxWidth' => 4096, 'maxHeight' => 4096]
+        ],
+        'uploadFolder' => '1:/user_upload/extbase_single_file/',
+    ])]
+
+Extbase will internally use the Extbase file upload validators for
+:php:`fileSize`, :php:`allowedMimeTypes` and :php:`imageDimensions` validation.
+
+Custom validators can be created according to project requirements and must
+extend the Extbase :php-short:`\TYPO3\CMS\Extbase\Validation\Validator\AbstractValidator`.
+The value to be validated is
+always a PSR-7 :php-short:`\TYPO3\CMS\Core\Http\UploadedFile` object.
+Custom validators can however not
+be used in the :php-short:`\TYPO3\CMS\Extbase\Annotation\FileUpload` attribute
+and must be configured manually.
+
+
+..  _extbase_fileupload_attribute-deletion:
+
+Deletion of uploaded files and file references
+----------------------------------------------
+
+The new Fluid ViewHelper
+:ref:`Form.uploadDeleteCheckbox ViewHelper <f:form.uploadDeleteCheckbox> <t3viewhelper:typo3-fluid-form-uploaddeletecheckbox>`
+can be used to show a "delete file" checkbox in a form.
+
+Example for object with :php-short:`\TYPO3\CMS\Core\Resource\FileReference` property:
+
+..  code-block:: php
+    :caption: Fluid code example
+
+    <f:form.uploadDeleteCheckbox property="file" fileReference="{object.file}" />
+
+Example for an object with an :php:`ObjectStorage<FileReference>` property,
+containing multiple files and allowing to delete the first one
+(iteration is possible within Fluid, to do that for every object of the collection):
+
+..  code-block:: php
+    :caption: Fluid code example
+
+    <f:form.uploadDeleteCheckbox property="file.0" fileReference="{object.file}" />
+
+Extbase will then handle file deletion(s) before persisting a validated
+object. It will:
+
+*   validate that minimum and maximum file upload configuration for the affected
+    property is fulfilled (only if the property has a :php-short:`\TYPO3\CMS\Extbase\Annotation\FileUpload`)
+*   delete the affected :php:`sys_file_reference` record
+*   delete the affected file
```

Comment:
What about the sys_file_processedfile entry and the processed files in _processed_ folder?

Reply:
IMO this is something for a Core review and hard to cover/discuss at this place. Processed files would IMO be cleaned up with the processed file cleanup automatism that people hopefully use. ;-)
Hope it's ok I'm resolving this one to move the doc patch forward :)